### PR TITLE
Allow key_node to be inside a nested struct.

### PR DESCRIPTION
**Pull request by ilyalesokhin-starkware**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1491)

### CODE REVIEW

**Review by spapinistarkware — CHANGES_REQUESTED** [review]
> Reviewed 2 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1491)* status: 2 of 3 files reviewed, 1 unresolved discussion (waiting on @ilyalesokhin-starkware)

___
*[`crates/syntax_codegen/src/generator.rs` line 162 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1491#-NKDYS014J9Jt0riXBZ7:-NKDYS022IJAq-xD91DV:b-1864c4) ([raw file](https://github.com/starkware-libs/cairo/blob/21a995c724604e130c9dcce60ce3777c0ced8bbe/crates/syntax_codegen/src/generator.rs#L162)):*
> ```XML
>                             .expect("Key node is missing from spec.");
> 
>                         match &key_node.kind {
> ```

This is a bit non intuitive.
Basically, when specifying key_node(), I get the key fields of this field. Only 2nd degree grand children. For each level.


I suggest either supporting grandchildren explicitly in the spec, and otherwise, fallback to old logic. e.g. .with_key("declaration.name").

**Review by ilyalesokhin-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/1491)* status: 2 of 3 files reviewed, 1 unresolved discussion (waiting on @spapinistarkware)

___
*[`crates/syntax_codegen/src/generator.rs` line 162 at r1](https://reviewable.io/reviews/starkware-libs/cairo/1491#-NKDYS014J9Jt0riXBZ7:-NKDhamp4M8Wkfq6pAf4:b-bmnu9o) ([raw file](https://github.com/starkware-libs/cairo/blob/21a995c724604e130c9dcce60ce3777c0ced8bbe/crates/syntax_codegen/src/generator.rs#L162)):*
<details><summary><i>Previously, spapinistarkware (Shahar Papini) wrote…</i></summary><blockquote>

This is a bit non intuitive.
Basically, when specifying key_node(), I get the key fields of this field. Only 2nd degree grand children. For each level.


I suggest either supporting grandchildren explicitly in the spec, and otherwise, fallback to old logic. e.g. .with_key("declaration.name").

</blockquote></details>

Can we make it a TODO and merge:
https://reviewable.io/reviews/starkware-libs/cairo/1489#-

without fixing it?

**Comment by ilyalesokhin-starkware** [comment]
The PR was put on hold as it is low priority.
